File: 1.py
def solve(arr: list) -> int:
    current_max = arr[0]
    max_sum = current_max

    for i in range(1, len(arr)):
        current_max = max((current_max + arr[i], arr[i]))
        max_sum = max((max_sum, current_max))

    return max_sum


if __name__ == "__main__":
    print(solve([4, -5, 2, 2, 2]))
    print(solve([2, 4, -2, -3, 8]))

File: 1.cpp
#include <iostream>
#include <vector>
#include <cmath>

using namespace std;

int solution(vector<int> arr)
{
    int currentSum = arr[0];
    int maxSum = currentSum;
    for (int i = 1; i < arr.size(); i++)
    {
        currentSum = max(currentSum + arr[i], arr[i]);
        maxSum = max(maxSum, currentSum);
    }
    return maxSum;
}

int main()
{
    vector<int> arr({3, -5, 2, 2, 2});
    int answer = solution(arr);
    cout << "The Answer is " << answer << endl;
    return 0;
}